### PR TITLE
add layer norm module for t5 model

### PR DESCRIPTION
**Pull request by pmabbo13**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #1829
* #1828
* #1827
* __->__ #1826
* #1833
* #1832
* #1831
* #1830
* #1825

### CODE REVIEW

**Review by Nayef211 — APPROVED** [review]

**Comment by pmabbo13** [comment]
# Description

This module perfrom Root-Mean-Square normalization (i.e. no subtraction of mean and no bias term). It is taken directly from [HF](https://github.com/huggingface/transformers/blob/8581a798c0a48fca07b29ce2ca2ef55adcae8c7e/src/transformers/models/t5/modeling_t5.py#L239)

File: torchtext/prototype/t5/modules.py [hunk 1]
```diff
@@ -418,3 +418,29 @@ def _relative_position_bucket(
...
+class T5LayerNorm(nn.Module):
+    def __init__(self, d_model, eps=1e-6):
+        """
+        Construct a layernorm module in the T5 style. No bias and no subtraction of mean.
+        """
+        super().__init__()
+        self.weight = nn.Parameter(torch.ones(d_model))
+        self.variance_epsilon = eps
+
+    def forward(self, hidden_states):
```
Comment (parmeet) [inline torchtext/prototype/t5/modules.py:433]: nit: Type annotate function and add doc string for hidden_states

**Review by parmeet — APPROVED** [review]
LGTM!
